packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -68,8 +75,12 @@ class Dialog extends Foundation<DialogHandledProps, DialogUnhandledProps, {}> {
      * React life-cycle method
      */
     public componentDidMount(): void {
-        if (canUseDOM() && this.props.onDismiss) {
-            window.addEventListener("keydown", this.handleWindowKeyDown);
+        if (canUseDOM()) {
+            document.addEventListener("keydown", this.handleDocumentKeyDown);
```

Comment:
Do we want to add the keydown event even if we don't need to listen to it (IE, there is no onDismiss handler, so escape does not close the dialog)

Reply:
I see - you're using this for the modal now as well.

Reply:
Restored similar logic to not listen when we don't need to.